### PR TITLE
Task context

### PR DESCRIPTION
**Pull request by ktlyee**

### CODE REVIEW

File: src/TaskContext.js [hunk 1]
```diff
@@ -46,8 +44,8 @@ export function TaskProvider({ children }) {
     }
     nextBoardId = allBoard[nextI];
 
-    const deleteone = newTasks[boardId].filter((task) => task.id !== id);
-    thisTasks[boardId] = [...deleteone]; // delete from current board
+    // delete from current board
+    thisTasks[boardId] = newTasks[boardId].filter((task) => task.id !== id);
   
     const moveTask = newTasks[boardId].filter((task) => task.id === id);
     thisTasks[nextBoardId].push(...moveTask) // add to des. board
```
Comment (Cijin) [inline src/TaskContext.js]: I think you are confused about the `...` operator
It's also called the spread operator. It's used to "spread" the values of an `Array` or `Object`

Use it when you want to create copies of something. In this instance it does not makes sense to use it.

Here `moveTask` is a single task, hence you could just say
```
thisTasks[next...].push(moveTask)
```

File: src/TaskContext.js [hunk 2]
```diff
@@ -21,18 +21,16 @@ export function TaskProvider({ children }) {
 
   const removeTask = (newTasks, boardId, id) => {
```
Comment (Cijin) [inline src/TaskContext.js:22]: When writing functions, follow this structure (till you find a better one :joy:)

```
/*
* @param newTasks: the task to be removed
* @param boardId: the board the task needs to be removed from
* @param id: the id of the task to be removed
*
* @returns: if something is returned, write that here
* 
* (next explain in one line what the function does)
* removes a task from a specific board
* 
* (in some cases provide some examples)
*/
const removeTask = (newTasks, boardId, id) => 
``` 

Just remember this structure the `next time` you write a function.

File: src/TaskContext.js (hunk 1 above)
Comment (ktlyee) [reply src/TaskContext.js]: when I remove `...` then I move task to other board its text doesn't show in a new board

Comment (Cijin) [reply src/TaskContext.js]: isn't `moveTask` of type `{ id, text}` ? Or is it something else?

Comment (pluemthnn) [reply src/TaskContext.js]: moveTask type is an object of array, like `[{ id, text}]`

**Review by Cijin — APPROVED** [review]

**Review by pluemthnn — APPROVED** [review]

**Pull request merged**